pkgs/development/libraries/glibc/check-etc-zoneinfo.patch

```diff
@@ -0,0 +1,36 @@
+diff --git a/time/tzfile.c b/time/tzfile.c
+index 040a5e34..a5a9c1c3 100644
+--- a/time/tzfile.c
++++ b/time/tzfile.c
+@@ -16,6 +16,7 @@
+    <https://www.gnu.org/licenses/>.  */
+ 
+ #include <assert.h>
++#include <errno.h>
+ #include <limits.h>
+ #include <stdio.h>
+ #include <stdio_ext.h>
+@@ -146,7 +147,22 @@ __tzfile_read (const char *file, size_t extra, char **extrap)
+ 	tzdir = default_tzdir;
+       if (__asprintf (&new, "%s/%s", tzdir, file) == -1)
+ 	goto ret_free_transitions;
+-      file = new;
++
++      // nixos specific adjustment - also check /etc/zoneinfo, but only when new
++      // equals TZDEFAULT and TZDEFAULT doesn't exist. In all other cases the
++      // user has asked for some other path and we can't default to /etc/zoneinfo.
```

Comment:
Why aren't we simply setting the default `TZDIR` value to `/etc/zoneinfo` in [`tzfile.h`](https://github.com/bminor/glibc/blob/595c22ecd8e87a27fd19270ed30fdbae9ad25426/timezone/tzfile.h#L25)?

Reply:
I don't even understand how this fixes the problem, since `TZDEFAULT` points to `/etc/localtime` and this file exists on my system when the bug occurs.

Reply:
first question: people are using nix on non-nixos machines where TZDIR is `/usr/share/zoneinfo`

second question: it's because I had yet another error in my thinking - we want to check `TZDIR` (the system default) against `tzdir` (just assigned) as a gating condition to check `/etc/zoneinfo`

I think it's correct now but going to spend a bit more time on it later.